src/core/ext/filters/client_channel/subchannel.cc

```diff
@@ -304,18 +304,24 @@ class ConnectedSubchannelStateWatcher
   static void OnHealthChanged(void* arg, grpc_error* error) {
     auto* self = static_cast<ConnectedSubchannelStateWatcher*>(arg);
     grpc_subchannel* c = self->subchannel_;
-    MutexLock lock(&c->mu);
-    if (self->health_state_ == GRPC_CHANNEL_SHUTDOWN) {
-      self->Unref();
-      return;
-    }
-    if (self->last_connectivity_state_ == GRPC_CHANNEL_READY) {
-      grpc_connectivity_state_set(&c->state_and_health_tracker,
-                                  self->health_state_, GRPC_ERROR_REF(error),
-                                  "health_changed");
+    bool unref = false;
```

Comment:
Instead of using a bool here, let's use the same approach we use in `OnConnectivityChanged()`: on line 320, add `self = nullptr`, and on line 324, check whether `self` is null before unreffing.

Reply:
Done.